vault_program/src/delegate_token_account.rs

```diff
@@ -0,0 +1,42 @@
+use jito_jsm_core::loader::{
+    load_associated_token_account, load_signer, load_token_mint, load_token_program,
+};
+use solana_program::{
+    account_info::AccountInfo, entrypoint::ProgramResult, program::invoke,
+    program_error::ProgramError, pubkey::Pubkey,
+};
+use spl_token::instruction::approve;
+
+pub fn process_delegate_token_account(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    amount: u64,
+) -> ProgramResult {
+    let [vrt_mint, token_account, owner, delegate, token_program] = accounts else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    load_associated_token_account(token_account, owner.key, vrt_mint.key)?;
+    load_token_mint(vrt_mint)?;
+    load_signer(owner, false)?;
+    load_token_program(token_program)?;
+
```

Comment:
you shall not be able to delegate the ATA for the supported mint out (otherwise vault admin can delegate the assets in the vault + rug)

Reply:
Added checking 
```rust
    if vault.supported_mint.eq(token_mint.key) {
        msg!("Invalid Token mint");
        return Err(ProgramError::InvalidAccountData);
    }
```